packages/start-slicemachine/src/lib/createSliceMachineServer.ts

```diff
@@ -0,0 +1,164 @@
+import {
+	SliceMachineManager,
+	createPrismicAuthManagerMiddleware,
+	createSliceMachineManagerMiddleware,
+} from "@slicemachine/manager";
+import { createServer, Server } from "node:http";
+import * as path from "node:path";
+import {
+	createApp,
+	fromNodeMiddleware,
+	toNodeListener,
+	sendProxy,
+	eventHandler,
+	createRouter,
+	proxyRequest,
+} from "h3";
+import serveStatic from "serve-static";
+import cors from "cors";
+import fetch from "node-fetch";
+import { createStaticFileEventHandler } from "./createStaticFileEventHandler";
+
+type CreateSliceMachineServerArgs = {
+	sliceMachineManager: SliceMachineManager;
+};
+
+/**
+ * Creates an HTTP server to handle the following:
+ *
+ * - Serve the Slice Machine app.
+ * - Expose a given Slice Machine manager to non-Node.js environments, like the
+ *   browser.
+ *
+ * The Slice Machine app is served from the project's Slice Machine module
+ * installation.
+ *
+ * If `NODE_ENV` is `development`, the Slice Machine app is served via a proxy
+ * to Next.js's development server. Slice Machine must be running on port 3000.
+ *
+ * @param args - Configuration for the server.
+ *
+ * @returns A standard `node:http` server.
+ */
+export const createSliceMachineServer = async (
+	args: CreateSliceMachineServerArgs,
+): Promise<Server> => {
+	const app = createApp();
+	const router = createRouter();
+
+	app.use(fromNodeMiddleware(cors()));
+
+	app.use(
+		"/_manager",
+		fromNodeMiddleware(
+			createSliceMachineManagerMiddleware({
+				sliceMachineManager: args.sliceMachineManager,
+			}),
+		),
+	);
+
+	app.use(
+		"/api/auth",
+		fromNodeMiddleware(
+			createPrismicAuthManagerMiddleware({
+				prismicAuthManager: args.sliceMachineManager.getPrismicAuthManager(),
+			}),
+		),
+	);
+
+	// TODO: Remove once tracking is implemented in `core2`
+	router.add(
+		"/api/s",
+		eventHandler(() => {
+			// noop
+
+			return {};
+		}),
+	);
+
+	if (process.env.NODE_ENV === "development") {
+		router.get(
+			"/**",
+			eventHandler((event) => {
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				const target = new URL(event.node.req.url!, "http://localhost:3000");
+
+				return proxyRequest(event, target.toString(), {
+					fetch: fetch as NonNullable<Parameters<typeof sendProxy>[2]>["fetch"],
```

Comment:
interesting that fetch needs to be injected

Reply:
`proxyRequest()` uses the global `fetch` function if it is available, which is only true on Node.js >= 18. Since we want to support Node.js as far back as at least 14, we need to provide it a `fetch`-compatible API, such as `node-fetch`.

(As of PR #852 [yet to be merged], h3 has been replaced by express, which removes this `proxyRequest()` function. Instead, we just use the `http-proxy-middleware` Express middleware.)